AutomatticTracks/src/main/java/com/automattic/android/tracks/crashlogging/CrashLogging.kt

```diff
@@ -1,7 +1,43 @@
 package com.automattic.android.tracks.crashlogging
 
 interface CrashLogging {
-    fun log(throwable: Throwable)
-    fun log(throwable: Throwable, data: Map<String, String?>)
-    fun log(message: String)
```

Comment:
A new interface for `CrashLogging` is inspired by current `WPAndroid` implementation. [You can check it here.](https://github.com/wordpress-mobile/WordPress-Android/blob/70ce7de8817020d29768074bb408688436320220/WordPress/src/main/java/org/wordpress/android/util/CrashLogging.kt#L87-L161)

An important thing for me when thinking of names was to make it easy for developers to **distinguish** which method _creates and sends_ an event vs. which method _adds something to breadcrumbs_ and doesn't send event **yet**.

I intentionally dropped word `log` as this is difficult to distinguish (to me) if it's about sending an event or not.

A request: @malinajirka could you please check names of new `CrashLogging` methods (and maybe docs above them) and tell if it was easy to understand what do they do and what's difference between them? I'm totally biased at this point so it's hard to tell for me. Thanks!